src/collection.rs

```diff
@@ -187,6 +187,18 @@ impl<T: Id<T>> CollectionWithId<T> {
             collection: self,
         }
     }
+
+    pub fn push(&mut self, item: T) -> Result<Idx<T>> {
+        let next_index = self.collection.objects.len();
+        let idx = Idx::new(next_index);
+        ensure!(
+            self.id_to_idx.insert(item.id().to_string(), idx).is_none(),
```

Comment:
It will put the collection in an inconsistent state.
```rust
let mut c = CollectionWithId::default();
c.push(Obj  { id: "foo".into() })).unwrap();
assert!(c.push(Obj  { id: "foo".into() }).is_err());
c[c.get_idx("foo").unwrap()]; // kaboom, the idx is invalid!
```

Hint: use the `entry` api.